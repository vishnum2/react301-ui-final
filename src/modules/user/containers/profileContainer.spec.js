import React from 'react';
import {shallow} from '../../../enzyme';
import {ProfileContainer, mapStateToProps} from './profileContainers';

describe('snapshot', () => {
    it('should do snapshot test', () => {
        const wrapper = shallow(<ProfileContainer userDetails={jest.fn()} detailsObject={{id: 'new'}}/>);
        expect(wrapper).toMatchSnapshot();
    })
});

describe('logged in check', () => {
    it('should render on login', () => {
        const wrapper = shallow(<ProfileContainer isLoggedIn={true} user={{balance_points: 12}}  userDetails={jest.fn()} detailsObject={{id: 'new'}}/>);
        expect(wrapper).toMatchSnapshot();
    })
})

describe('check mapstate', () => {
    it('should check mapstatetoprops', () => {
        const state = {
            login: {
                loginStatus: true,
                detailsObject: []
            },
            users: {
                userDetails: [{
                    new: 'user'
                }]
            }
        }
        const response = {
            isLoggedIn: state.login.loginStatus,
            detailsObject: state.login.detailsObject,
            user: state.users.UserDetails
        }
        expect(mapStateToProps(state)).toEqual(response);
    })
})
