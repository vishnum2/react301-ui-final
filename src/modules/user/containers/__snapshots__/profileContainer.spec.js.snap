// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logged in check should render on login 1`] = `ShallowWrapper {}`;

exports[`snapshot should do snapshot test 1`] = `ShallowWrapper {}`;
